### PR TITLE
UI-9249 - Migrate default selected drillthrough columns

### PR DESCRIPTION
**Pull request by Nouzbe**
The migration script used to disregard default selected drillthrough columns configured in UI 4.3 when it is used to migrate the content to a more recent version.

This PR takes care of it and makes sure that this configuration is not lost.

### CODE REVIEW

File: src/4.3_to_5.0/migrateSettingsFolder.ts [hunk 1]
```diff
@@ -38,6 +39,43 @@ function migrateSettingsMap(legacySettingsMap: {
...
+        );
+      }
+    }
+
+    // Remove duplicates.
+    migratedSettingsMap["drillthrough.defaultSelectedColumns"] = _mapValues(
+      migratedDrillthroughColumns,
+      _uniq,
+    );
+  }
```
Comment (messaooudi) [inline src/4.3_to_5.0/migrateSettingsFolder.ts:77]: This implies that two drillthrough columns with the same `columnName`  but different `functionName` are considered duplicates. Right ?

File: src/4.3_to_5.0/legacySettingsFolder.ts [hunk 2]
```diff
@@ -14,6 +14,58 @@ const user1Preferences = {
...
+        ],
+        SecondCube: [
+          {
+            functionName: "MemberValue",
+            columnName: "delta",
+          },
+          {
+            functionName: "Caption",
+            columnName: "delta",
+          },
```
Comment (messaooudi) [inline src/4.3_to_5.0/legacySettingsFolder.ts:45]: This is considered duplicates, so we are testing that part of the implementation ?

**Review by messaooudi — COMMENTED** [review]
Looks good, I have one question to make sur i understand the remove duplicate part.

File: src/4.3_to_5.0/migrateSettingsFolder.ts (hunk 1 above)
Comment (Nouzbe) [reply src/4.3_to_5.0/migrateSettingsFolder.ts:77]: Indeed. In fact there are even two ways to have duplicates:
- because UI4 was typically storing the following for each column:
```
{
  functionName: "MemberValue",
  columnName: "myColumn",
}, {
  functionName: "MemberCaption",
  columnName: "myColumn"
}
```
So we don't want to save each column twice in the new Atoti UI 5 setting.

The other way is more rare. It is because UI4 was saving the columns per server and per cube whereas UI5 only stores them per cube. So you could have a duplicate because a column was save for a cube with the same name under two different servers.

The current implementation in this PR should get rid of any duplicate, whether they come from the first way or the second one I described here.

File: src/4.3_to_5.0/legacySettingsFolder.ts (hunk 2 above)
Comment (Nouzbe) [reply src/4.3_to_5.0/legacySettingsFolder.ts:45]: This is considered duplicates indeed, but it's the first way to have duplicate I described above. Technically, I didn't test the second way of having duplicates.

I also tested the second way of having duplicates, as I added the column `subTradeId` in `MarketRiskCube` under two different servers.

**Review by messaooudi — APPROVED** [review]
Thanks for the explanations. LGTM
